tests/reveal/ndarray_conversion.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+
+nd = np.array([[1, 2], [3, 4]])
+
+# item
+reveal_type(nd.item())  # E: Any
+reveal_type(nd.item(1))  # E: Any
+reveal_type(nd.item(0, 1))  # E: Any
+reveal_type(nd.item((0, 1)))  # E: Any
+
+# tolist
+reveal_type(nd.tolist())  # E: builtins.list[Any]
+
+# itemset does not return a value
+# tofile does not return a value
+# dump does not return a value
+
+# astype
+reveal_type(nd.astype("float"))  # E: numpy.ndarray
+reveal_type(nd.astype(float))  # E: numpy.ndarray
+
+reveal_type(nd.astype(float, "K"))  # E: numpy.ndarray
+reveal_type(nd.astype(float, order="K"))  # E: numpy.ndarray
+
+reveal_type(nd.astype(float, "K", "unsafe"))  # E: numpy.ndarray
+reveal_type(nd.astype(float, casting="unsafe"))  # E: numpy.ndarray
+
+reveal_type(nd.astype(float, "K", "unsafe", True))  # E: numpy.ndarray
+reveal_type(nd.astype(float, subok=True))  # E: numpy.ndarray
+
+reveal_type(nd.astype(float, "K", "unsafe", True, True))  # E: numpy.ndarray
+reveal_type(nd.astype(float, copy=True))  # E: numpy.ndarray
+
+# byteswap
+reveal_type(nd.byteswap())  # E: numpy.ndarray
+reveal_type(nd.byteswap(True))  # E: numpy.ndarray
+
+# copy
+reveal_type(nd.copy())  # E: numpy.ndarray
+reveal_type(nd.copy("C"))  # E: numpy.ndarray
+
+# view
+reveal_type(nd.view())  # E: numpy.ndarray
+reveal_type(nd.view(np.int64))  # E: numpy.ndarray
+reveal_type(nd.view(dtype=np.int64))  # E: numpy.ndarray
+reveal_type(nd.view(np.int64, np.matrix))  # E: numpy.ndarray
+reveal_type(nd.view(type=np.matrix))  # E: numpy.ndarray
```

Comment:
In theory, I think this should be `numpy.matrix`.

Or we can could make a dummy subclass, if `np.matrix` isn't defined, e.g.,
```
class SubArray(np.ndarray):
    pass

reveal_type(nd.view(type=SubArray))  # E: SubArray
```

Does that actually work?

Reply:
Thank you for review! I've removed simple tests and fixed type hints for `view`.